third_party/asr/asr582x/BUILD.gn

```diff
@@ -0,0 +1,382 @@
+      "${asr_sdk_build_root}/asr_sdk/lwip/lwip_2.1.2/port/include",
+      "${asr_sdk_build_root}/asr_sdk/lwip/lwip_2.1.2/include/lwip",
+      "${asr_sdk_build_root}/asr_sdk/lwip/lwip_app",
+      "${asr_sdk_build_root}/asr_sdk/lwip/lwip_app/lwip_app_ping",
+      "${asr_sdk_build_root}/asr_sdk/lwip/lwip_2.1.2/include/lwip/prot",
+      "${asr_sdk_build_root}/asr_sdk/lwip/lwip_2.1.2/include/netif",
+      "${asr_sdk_build_root}/asr_sdk/lwip/lwip_app/iperf",
+      "${asr_sdk_build_root}/asr_sdk/lwip/lwip_app/ota",
+      "${asr_sdk_build_root}/asr_sdk/lwip/mbedtls-mbedtls-2.18/include",
+      "${asr_sdk_build_root}/asr_sdk/lwip/mbedtls-mbedtls-2.18/include/mbedtls",
+      "${asr_sdk_build_root}/asr_sdk/lwip/mbedtls-mbedtls-2.18/asr_mbedtls/",
+      "${asr_sdk_build_root}/asr_sdk/lwip/mbedtls-mbedtls-2.18/asr_mbedtls/alt",
+      "${asr_sdk_build_root}/asr_sdk/demo/matter_demo/app/ble_demo/duet_hl_matter",
+      "${asr_sdk_build_root}/asr_sdk/platform/duet/kv/include",
+  ]
+
+  # Also add project include path (passed in from project build)
+  proj_path = get_label_info(asr_target_project, "dir")
+  _include_dirs += [ "/" + rebase_path("${proj_path}/include", "/", ".") ]
+
+  cflags = []
+  foreach(include_dir, _include_dirs) {
+    cflags += [ "-isystem" + rebase_path(include_dir, root_build_dir) ]
+  }
+
+  # Pull out defines from generated json
+  defines = [
+    "_SPI_FLASH_240MHz_",
+    "DCDC_PFMMODE_CLOSE",
+    "LEGA_A0V2",
+    "DUET_CM4",
+    "_SPI_FLASH_ENABLE_",
+    "SYSTEM_SUPPORT_OS",
+    "WIFI_DEVICE",
+    "CFG_BATX=1",
+    "CFG_BARX=1",
+    "CFG_REORD_BUF=4",
+    "CFG_SPC=4",
+    "CFG_TXDESC0=4",
+    "CFG_TXDESC1=4",
+    "CFG_TXDESC2=4",
+    "CFG_TXDESC3=4",
+    "CFG_TXDESC4=4",
+    "CFG_CMON",
+    "CFG_MDM_VER_V21",
+    "CFG_SOFTAP_SUPPORT",
+    "CFG_SNIFFER_SUPPORT",
+    "CFG_CUS_FRAME",
+    "AT_USER_DEBUG",
+    "CFG_WF_DBG=1",
+    "__FPU_PRESENT=1",
+    "DX_CC_TEE",
+    "HASH_SHA_512_SUPPORTED",
+    "CC_HW_VERSION=0xF0",
+    "DLLI_MAX_BUFF_SIZE=0x10000",
+    "SSI_CONFIG_TRNG_MODE=0",
+    "LWIP_APP_IPERF",
+    "SYSTEM_RECOVERY",
+    "LWIP_DUALSTACK",
+    "SYSTEM_COREDUMP",
+    "CFG_REC",
+    "CFG_SNIFFER_UNICAST_SUPPORT",
+    "CFG_OS_VERSION_CHECK",
+    "PS_CLOSE_APLL",
+    "CFG_STA_AP_COEX_N",
+    "CFG_SOFTAP_SWITCH_CHAN_N",
+    "CFG_PLF_DUET",
+    "SONATA_RTOS_SUPPORT",
+    "CFG_SEG",
+    "CFG_DUET_FREERTOS",
+    "PRINTF2_SUPPORT",
+    "DUET_RF_SLEEP",
+    "CFG_DUET_BLE",
+    "CFG_RTC_INDEPENDENT",
+    "CFG_SAE",
+    "CFG_SAE_AUTH_OPT",
+    "CFG_MFP",
+    "CONFIG_IEEE80211W",
+    "WFA_CERTIFICATE_N",
+    "CFG_DUET_5822S",
+    "CFG_DUET_5822T",
+    "MBEDTLS_HW",
+    "MBEDTLS_HW_RSA",
+    "MBEDTLS_HW_ECC",
+    "MBEDTLS_HW_AES",
+    "MBEDTLS_HW_HASH",
+    "MBEDTLS_USER_CONFIG_FILE=<mbedtls_user_config.h>",
+    "CONFIG_ENABLE_ASR_LEGA_RTOS=1",
+#    "CONFIG_ENABLE_ASR_APP_MESH=1",
+    "CFG_EASY_LOG_ENABLE=1",
+    "CFG_EASY_LOG_MODULE_EN=1",
+    ]
+
+  cflags_c = [
```

Comment:
could we somehow use a single declaration and re-use settings for c,cc,asm to not repeat the same list?